### PR TITLE
typescript and other dependences and modules added to package.json

### PR DESCRIPTION
**Pull request by BenMorant**
I followed https://stackoverflow.com/questions/47508564/migrating-create-react-app-from-javascript-to-typescript =>

$ npm install --save typescript @types/node @types/react @types/react-dom @types/jest

Some typescript-related modules and dependencies have been mentionned in package.json . 

However, if I change all the .js to .tsx, I 1) have lots of errors and 2) can't launch npm start .

Anyway, if I started a new project with TS pre-enabled, I would type  npx create-react-app my-app --typescript , but no .ts nor .tsx would be created .

So... did I really configure the app so it can use typescript ?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,19 +1,48 @@
...
+  },
+  "watch": {
+    "test": "{src,test}/*.js"
   },
-  "devDependencies": {},
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
-    "test": "react-scripts test --env=jsdom",
+    "test": "tape test/*.js",
```
Comment (xavierhardy) [inline package.json:26]: `jest` that was used before works with TypeScript

File: package.json [hunk 2]
```diff
@@ -26,5 +55,21 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
+  },
+  "devDependencies": {
+    "husky": "^4.2.5",
+    "lint-staged": "^10.2.11",
+    "prettier": "2.0.5"
+  },
```
Comment (xavierhardy) [inline package.json:63]: prettier has nothing to do with TypeScript

File: package.json [hunk 3]
```diff
@@ -26,5 +55,21 @@
...
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "*.{json,txt,ts,tsx,jsx,html,js,css,md}": ["prettier --write",
+    "eslint --fix",
+    "git add",
+    "jest --bail --findRelatedTests"]
```
Comment (xavierhardy) [inline package.json:73]: Are we using Jest or Tape? Why is there a `git add` in the middle of this.

File: public/index.html [hunk 4]
```diff
@@ -24,8 +36,8 @@
       You can add webfonts, meta tags, or analytics to this file.
       The build step will place the bundled scripts into the <body> tag.
 
-      To begin the development, run `npm start`.
-      To create a production bundle, use `npm run build`.
+      To begin the development, run `npm start` or `yarn start`.
+      To create a production bundle, use `npm run build` or `yarn build`.
```
Comment (xavierhardy) [inline public/index.html:40]: None of these changes have anything to do with TypeScript. The same goes for the images that have been replaced/removed.

Comment (BenMorant) [reply public/index.html:40]: It has to do with the generation of the TypeScript stuff (I finally did it, but it was a pain in the ass). 
I'll try to replace the images

File: package.json (hunk 3 above)
Comment (BenMorant) [reply package.json:73]: Removing....

File: package.json (hunk 2 above)
Comment (BenMorant) [reply package.json:63]: Removing....

File: package.json (hunk 1 above)
Comment (BenMorant) [reply package.json:26]: should be better now

File: public/index.html (hunk 4 above)
Comment (BenMorant) [reply public/index.html:40]: done...

**Pull request merged**